ci-operator/step-registry/cucushift/installer/rehearse/aws/cases/clusters/provision/cucushift-installer-rehearse-aws-cases-clusters-provision-commands.sh

```diff
@@ -0,0 +1,741 @@
+
+  EDGE_ZONE_COUNT=$(jq -r '. | length' ${EDGE_ZONES_JSON})
+  EDGE_ZONE_NAMES=$(jq -r '.[].ZoneName' ${EDGE_ZONES_JSON})
+
+  if [ "${ARCH}" == "" ]; then
+    yq-v4 eval -i '.compute[1].architecture = "amd64"' "${CONFIG}"
+  else
+    export ARCH
+    yq-v4 eval -i '.compute[1].architecture = env(ARCH)' "${CONFIG}"
+  fi
+
+  yq-v4 eval -i '.compute[1].hyperthreading = "Enabled"' "${CONFIG}"
+  yq-v4 eval -i '.compute[1].name = "edge"' "${CONFIG}"
+  export EDGE_ZONE_COUNT
+  yq-v4 eval -i '.compute[1].replicas = env(EDGE_ZONE_COUNT)' "${CONFIG}"
+  for edge_zone in ${EDGE_ZONE_NAMES}; do
+    export edge_zone
+    yq-v4 eval -i '.compute[1].platform.aws.zones += [env(edge_zone)]' "${CONFIG}"
+    unset edge_zone
+  done
+fi
+
+echo "install-config.yaml:"
+yq-v4 '({"compute": .compute, "controlPlane": .controlPlane, "platform": .platform})' "${CONFIG}"
+
+# ---------------------------------------
+
+echo "--- Create manifests ---"
+
+set +e
+openshift-install create manifests --dir ${INSTALL_DIR} &
+wait "$!"
+install_ret="$?"
+set -e
+
+ret=$((ret + install_ret))
+if [ $install_ret -ne 0 ]; then
+  echo "Failed to create manifests. Exit code: $install_ret"
+  INSTALL_RESULT="FAIL"
+else
+  echo "Created manifests."
+fi
+
+# ---------------------------------------
+
+echo "--- Create ignition configs ---"
+
+set +e
+openshift-install create ignition-configs --dir ${INSTALL_DIR} &
+wait "$!"
+install_ret="$?"
+set -e
+
+ret=$((ret + install_ret))
+if [ $install_ret -ne 0 ]; then
+  echo "Failed to ignition configs. Exit code: $install_ret"
+  INSTALL_RESULT="FAIL"
+else
+  echo "Created ignition configs."
+fi
+
+# ---------------------------------------
+
+echo "--- Create cluster ---"
+
+set +e
+openshift-install create cluster --dir ${INSTALL_DIR} 2>&1 | grep --line-buffered -v 'password\|X-Auth-Token\|UserData:' &
+wait "$!"
+install_ret="$?"
+set -e
+
+if [ $install_ret -ne 0 ]; then
+  echo "Failed to create clusters. Exit code: $install_ret"
+  INSTALL_RESULT="FAIL"
+else
+  echo "Created cluster."
+  INSTALL_RESULT="PASS"
+fi
+ret=$((ret + install_ret))
+
+# ---------------------------------------
+# Health check
+# ---------------------------------------
+
+echo "--- Health check ---"
+if [ $install_ret -eq 0 ]; then
+
+  if [[ -f ${INSTALL_DIR}/auth/kubeconfig ]]; then
+    export KUBECONFIG=${INSTALL_DIR}/auth/kubeconfig
+
+    health_check_ret=0
+
+    set +e
+    health_check "${REGION}"
+    set -e
+
+    health_check_ret="$?"
+    
+    if [ $health_check_ret -ne 0 ]; then
+      echo "Healthcheck failed."
+      HEALTHCHECK_RESULT="FAIL"
+      ret=$((ret + 1))
+    else
+      echo "Healthcheck succeeded."
+      HEALTHCHECK_RESULT="PASS"
+    fi
+  else
+    echo "No kubeconfig found."
+    ret=$((ret + 1))
+  fi
+else
+  echo "Install FAIL, skip health check."
+  HEALTHCHECK_RESULT="FAIL"
+fi
+
+
+# ---------------------------------------
+# Gather must-gather
```

Comment:
Can we directly insert must-gather step in the provision chain to save the codes here ?